docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md

```diff
@@ -245,3 +245,22 @@ with `cert-manager.io/v1`
   ```yaml
   serviceAccountName: manager
   ```
+
+## Required change to support externally managed infrastructure.
+
+- A new annotation `cluster.x-k8s.io/managed-by` has been introduced that allows cluster infrastructure to be managed
+externally.
+- When this annotation is added to an `InfraCluster` resource, the controller for these resources should not reconcile
+the resource.
+- The `ResourceNotManagedExternally` predicate is a useful helper to check for the annotation and the filter the resource easily:
+  ```go
+  c, err := ctrl.NewControllerManagedBy(mgr).
+		For(&providerv1.InfraCluster{}).
+		Watches(...).
+		WithOptions(options).
+		WithEventFilter(predicates.ResourceNotManagedExternally(ctrl.LoggerFrom(ctx))).
```

Comment:
Hm the question really is, what are the alternatives. Not sure if `ResourceManagedByUs` or something similar to get rid of the `not` would be better.

Reply:
I was following the `ResourceNotPaused` approach for the naming here. IMO the problem is the same for that one, "with event filter resource not paused". Happy to rename if we can think of something better.

- ResourceIsManaged
- ResourceInternallyManaged

🤔

Reply:
looking at: https://cs.k8s.io/?q=WithEventFilter&i=nope&files=&excludeFiles=&repos= 
Sometimes the predicates are also named in this style: `WithEventFilter(resource.NewPredicates(isInOperatorNamespace)).` Ideally we shorten that even further.

So in our case this could be one of:
* WithEventFilter(resource.IsNotManagedExternally)
* WithEventFilter(resource.IsManagedByUs)
* WithEventFilter(resource.IsManaged)
* WithEventFilter(resource.IsInternallyManaged)

I like about that that it does move the "Resource" in the package path, which removes the "useless" `predicates`. Not sure if I like the implications on the go package names and the deprecations this requires.

Reply:
From above `WithEventFilter(resource.IsNotManagedExternally)`, I would tweak it a bit with `WithEventFilter(resource.IsNotExternallyManaged)`? That reads much better, thoughts?

Reply:
I recommended https://github.com/kubernetes-sigs/cluster-api/pull/4303#discussion_r618636981 to turn it around because we already had `IsManagedExternally`. I think we should keep both consistent. English is not my first language, but to me `WithEventFilter(resource.IsNotExternallyManaged)` also sounds more fluent.

Reply:
Yeah we had a mix of `ExternallyManaged` and `ManagedExternally` so I made that consistent after @sbueringer highlighted, I don't have a strong opinion on which direction is better though, they both work IMO. Happy to switch the other way but if we do I'll do it across all the places where we have these two words together.

Note the proposal was always "externally managed", so maybe we want to go that way? `Managed Externally Infrastructure` doesn't work, `Externally Managed Infrastructure` does 🤔

Reply:
Maybe let's just do `ResourceIsNotExternallyManaged`? Reads better with the `is` in it